lib/InvalidArgumentError.js

```diff
@@ -0,0 +1,9 @@
+var CustomError = require("./CustomError.js");
+var util = require("util");
+
+function InvalidArgumentError(message, previousError) {
```

Comment:
Would be really cool if first arg was the name of the argument that was invalid:

`function InvalidArgumentError(argumentName, message, previousError) {`